### PR TITLE
Implement `plot_parallel_coordinate` and `_get_parallel_coordinate_plot` with Matplotlib backend.

### PR DESCRIPTION
**Pull request by ytknzw**
This PR proposes to implement a ``matplotlib`` backend for plotting parallel coordinate based on PR #1756 .

## Motivation
To provide another visualization option for users, who have only ``plotly`` now.
See issue #1539 and PR #1707 for the original motivation.

## Description of the changes
Create  ``optuna.visualization.matplotlib._parallel_coordinate.plot_parallel_coordinate`` and ``optuna.visualization.matplotlib._parallel_coordinate._get_parallel_coordinate_plot``.

### CODE REVIEW

**Comment by ytknzw** [comment]
The commit https://github.com/optuna/optuna/pull/1764/commits/8a4a8d830755a40c192750ee2895ada57bd771ee fixes the Issue #1809 .

File: optuna/visualization/matplotlib/_parallel_coordinate.py [hunk 1]
```diff
@@ -1,11 +1,130 @@
...
+        for input_p_name in params:
+            if input_p_name not in all_params:
+                raise ValueError("Parameter {} does not exist in your study.".format(input_p_name))
+        all_params = set(params)
+    sorted_params = sorted(list(all_params))
+
+    obj_org = [t.value for t in trials]
+    obj_min = min(obj_org)
+    obj_max = max(obj_org)
+    obj_w = obj_max - obj_min  # type: ignore
```
Comment (crcrpar) [inline optuna/visualization/matplotlib/_parallel_coordinate.py:72]: [question] why is the type ignored?

File: optuna/visualization/matplotlib/_parallel_coordinate.py [hunk 2]
```diff
@@ -1,11 +1,130 @@
...
+            cat_param_values.append([v[0] for v in vocab_item_sorted])
+            cat_param_ticks.append([v[1] for v in vocab_item_sorted])
+
+        p_min = min(values)
+        p_max = max(values)
+        p_w = p_max - p_min
+        for i, v in enumerate(values):
+            dims_obj_base[i].append((v - p_min) / p_w * obj_w + obj_min)
+
+        var_names.append(p_name if len(p_name) < 20 else "{}...".format(p_name[:17]))
```
Comment (crcrpar) [inline optuna/visualization/matplotlib/_parallel_coordinate.py:98]: [question] why is the name truncated at 17?

File: optuna/visualization/matplotlib/_parallel_coordinate.py [hunk 3]
```diff
@@ -1,11 +1,130 @@
...
+        param_values.append(values)
+
+    # Draw multiple line plots and axes.
+    # https://stackoverflow.com/a/50029441
```
Comment (crcrpar) [inline optuna/visualization/matplotlib/_parallel_coordinate.py:102]: [nit-picking]
```suggestion
    # Ref: https://stackoverflow.com/a/50029441
```

**Review by crcrpar — COMMENTED** [review]
Thank you for your work.
It generally looks good but I have some questions and nit-pickings 🙂 
image

File: optuna/visualization/matplotlib/_parallel_coordinate.py (hunk 2 above)
Comment (ytknzw) [reply optuna/visualization/matplotlib/_parallel_coordinate.py:98]: This one is based on the Plotly backend.
https://github.com/optuna/optuna/blob/f6ebb4c640164b5bece2357cfd933610428c71b5/optuna/visualization/_parallel_coordinate.py#L105

File: optuna/visualization/matplotlib/_parallel_coordinate.py (hunk 3 above)
Comment (ytknzw) [reply optuna/visualization/matplotlib/_parallel_coordinate.py]: Thank you for reviewing! Just added the term.

File: optuna/visualization/matplotlib/_parallel_coordinate.py (hunk 1 above)
Comment (ytknzw) [reply optuna/visualization/matplotlib/_parallel_coordinate.py:72]: I put this temporalily but forgot to remove it before submitting PR, sorry.

Comment (ytknzw) [reply optuna/visualization/matplotlib/_parallel_coordinate.py]: Corrected.

File: optuna/visualization/matplotlib/_parallel_coordinate.py (hunk 2 above)
Comment (crcrpar) [reply optuna/visualization/matplotlib/_parallel_coordinate.py:98]: Ah, got it. Thanks 😃

File: optuna/visualization/matplotlib/_parallel_coordinate.py (hunk 1 above)
Comment (crcrpar) [reply optuna/visualization/matplotlib/_parallel_coordinate.py]: okay, thanks.

**Review by crcrpar — APPROVED** [review]

**Review by himkt — APPROVED** [review]
@ytknzw 

Really sorry for the late. And thank you for the great contribution.
I went through the PR and tried on some examples. LGTM

**Comment by ytknzw** [comment]
Thank you!

**Pull request merged**